proto/message.py

```diff
@@ -685,7 +722,24 @@ def __setattr__(self, key, value):
         if key[0] == "_":
             return super().__setattr__(key, value)
         marshal = self._meta.marshal
-        pb_type = self._meta.fields[key].pb_type
+        try:
+            pb_type = self._meta.fields[key].pb_type
+        except KeyError:
```

Comment:
These three newly introduced sections seem very similar (and the comment section too). Can we put it in a helper method or something to avoid code duplication? The small differences can be addressed by passing some sort of parameters into a helper funciton?